### DISCUSSION TITLE
How to replace text tracks dynamically without changing persisted captions state in local storage?

### DISCUSSION THREAD

**Discussion by DerStimmler** [vidstack/player#1126] [Q&A] [answered]
Hi, I'd like to know how I can dynamically replace text tracks without changing the persisted `captions` state in local storage.

Currently I activated storage support by adding the `storage` attribute to the `media-player` component:
```
<media-player
    #player
    playsinline
    load="visible"
    view-type="video"
    stream-type="on-demand"
    storage="media-player"
    ...
```

When I enable captions that change is reflected in local storage:
```
{
  "volume": 0.47,
  "muted": false,
  "lang": "DE",
  "captions": true,
  "rate": 1.25
}
```

Now when I change the video source and therefore the text tracks dynamically with this code:
```
this.player.src = {
        src: streamUrl,
        type: 'application/mpegurl'
      };

      this.player.textTracks.clear();

      captions.forEach(caption => {
        const track = new TextTrack({
          src: caption.source,
          language: caption.language,
          type: 'vtt',
          kind: 'subtitles',
          label: caption.label
        });

        this.player.textTracks.add(track);
      });
```

The `captions` state in local storage is set to `false` because I first cleared all text tracks.
```
{
    "volume": 0.47,
    "muted": false,
    "lang": "DE",
    "captions": false,
    "rate": 1.25
}
```

I also tried to first add the new text tracks and remove the old ones afterwards but the same result:
```
      const textTracksBefore = this.player.textTracks.toArray();

      captions.forEach(caption => {
        const track = new TextTrack({
          src: caption.source,
          language: caption.language,
          type: 'vtt',
          kind: 'subtitles',
          label: caption.label
        });

        this.player.textTracks.add(track);
      });

      textTracksBefore.forEach(track => {
        this.player.textTracks.remove(track);
      });
```

I there a proper way to do this?

**Comment by mihar-22** [comment, marked as answer]
This is a bug and I've added a fix. It'll be rolling out in `1.10.4@next` today or tomorrow. Thanks for letting me know :)

**Comment by MrVibe** [comment]
is this released yet  ? in react version I am facing same issue and chapters do not work
